### ISSUE TITLE
Loading json via the url: scheme stopped working with 2.0.0-nightly.733

### ISSUE THREAD

**Issue by usefulthink** [parcel-bundler/parcel#7151]
# 🐛 bug report

I'm not sure if this is new intended behaviour and I'm doing it wrong, but this used to work and doesn't now.

As described in the title, with all versions after 2.0.0-nightly.733 (release timestamp 2021-06-25T00:36:11.674Z) I am no longer able to load a large json-file using the `url:`-Scheme.

## 🎛 Configuration (.babelrc, package.json, cli command)

No configuration. 

```
import jsonUrl from 'url:./test.json';

console.log(jsonUrl);
```

full example can be found here:

https://gist.github.com/usefulthink/8798763f160f0966b76242c73f725ae1

## 🤔 Expected Behavior

I would expect parcel to place a json-file into the ./dist-folder when building and return its url (this worked fine up until 2.0.0-nightly.732)

## 😯 Current Behavior

The file gets transformed into a JS-file and a .js-URL is returned. `(await fetch(jsonUrl)).json()` doesn't like that.

## 🔦 Context

I first determined the first version this issue appeared in (2.0.0-nightly.733) and checked the changes around the release-date and it appears that https://github.com/parcel-bundler/parcel/commit/dee58e9721c535a2431d734388cd6b9e319e5c7c#diff-3ca400b6b91bc49c60c3c75951d144849778233b303de0a706863daa69bc50d6R42 might be responsible for this change in behaviour (specifically the change to the default-handling of `'url:*'`-dependencies).

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           | 2.0.0-nightly.733 - 2.0.0
| Node             | 16.9.1
| npm/Yarn         |   7.21.1
| Operating System | macOS 10.15

**Comment by usefulthink** [comment]
It appears that the following `.parcelrc` restores the previous behaviour. Is that correct?

```json
{
  "extends": "@parcel/config-default",
  "transformers": {
    "url:*.json": ["@parcel/transformer-raw"]
  }
}
```

are there any known cases where transforming a JSON-file imported this way into JS is actually desired? 
Otherwise, would adding this transformer declaration to the default-configuration make sense? I will happily provide a PR for this if you like.

**Comment by devongovett** [comment]
This is a larger issue that there's no way to change the behavior of how a file is transformed depending on where it is imported. In JavaScript, you want to get back an object generally, but when referencing JSON in HTML or via a URL, you want actual JSON. For now, creating a config like above, or using a different named pipeline (e.g. `json:`) is the best way to handle this.

**Comment by usefulthink** [comment]
Yeah, I did go with a pipeline `raw-url:` (using `@parcel/transformer-raw`) that I can use for all files where I don't want any processing to happen. Would it make sense to add something like that to the default-configuration?
